source/common/tls/context_impl.cc

```diff
@@ -212,6 +212,14 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
       if (!creation_status.ok()) {
         return;
       }
+
+      // Create and set the certificate expiration gauge.
+      ctx.createCertExpirationGauge(scope, tls_certificate.certificateName());
+      if (ctx.cert_chain_.get() != nullptr) {
+        ctx.setExpirationOnCertStats(
+            std::chrono::seconds(Utility::getExpirationUnixTime(ctx.cert_chain_.get())));
+      }
```

Comment:
I don't think you need two extra functions for this; just inline the code. Also, I don't think you need to a member to keep the pointer to the gauge; it is never used again. And let's also move the "there isn't a cert" case into `getExpirationUnixTime` to be the same as `getDaysUntilExpiration` in that regard.

```suggestion
      Stats::Gauge& expire = Extensions::TransportSockets::Tls::createCertificateExpirationGauge(scope, cert_name);
      expire.set(std::chrono::seconds(Utility::getExpirationUnixTime(ctx.cert_chain_.get())).count());
```